packages/x-date-pickers-pro/src/EnhancedDateRangePickerDay/EnhancedDateRangePickerDay.tsx

```diff
@@ -0,0 +1,661 @@
+    showDaysOutsideCurrentMonth,
+  });
+
+  const ownerState: EnhancedDateRangePickerDayOwnerState = {
+    ...pickersDayOwnerState,
+    // Properties that the Base UI implementation will have
+    isDaySelectionStart: isStartOfHighlighting,
+    isDaySelectionEnd: isEndOfHighlighting,
+    isDayInsideSelection: isHighlighting && !isStartOfHighlighting && !isEndOfHighlighting,
+    isDaySelected: isHighlighting || Boolean(selected),
+    isDayPreviewed: isPreviewing,
+    isDayPreviewStart: isStartOfPreviewing,
+    isDayPreviewEnd: isEndOfPreviewing,
+    isDayInsidePreview: isPreviewing && !isStartOfPreviewing && !isEndOfPreviewing,
+    // Properties specific to the MUI implementation (some might be removed in the next major)
+    isDayStartOfMonth: utils.isSameDay(day, utils.startOfMonth(day)),
+    isDayEndOfMonth: utils.isSameDay(day, utils.endOfMonth(day)),
+    isDayFirstVisibleCell: isFirstVisibleCell,
+    isDayLastVisibleCell: isLastVisibleCell,
+    isDayFillerCell: outsideCurrentMonth && !showDaysOutsideCurrentMonth,
+    isDayDraggable: Boolean(draggable),
+  };
+
+  const classes = useUtilityClasses(ownerState);
+
+  const ref = React.useRef<HTMLButtonElement>(null);
+  const handleRef = useForkRef(ref, forwardedRef);
+
+  // Since this is rendered when a Popper is opened we can't use passive effects.
+  // Focusing in passive effects in Popper causes scroll jump.
+  useEnhancedEffect(() => {
+    if (autoFocus && !disabled && !isAnimating && !outsideCurrentMonth) {
+      // ref.current being null would be a bug in MUI
+      ref.current!.focus();
+    }
+  }, [autoFocus, disabled, isAnimating, outsideCurrentMonth]);
+
+  // For a day outside the current month, move the focus from mouseDown to mouseUp
+  // Goal: have the onClick ends before sliding to the new month
+  const handleMouseDown = (event: React.MouseEvent<HTMLButtonElement>) => {
+    onMouseDown(event);
+    if (outsideCurrentMonth) {
+      event.preventDefault();
+    }
+  };
+
+  const handleClick = (event: React.MouseEvent<HTMLButtonElement>) => {
+    if (!disabled) {
+      onDaySelect(day);
+    }
+
+    if (outsideCurrentMonth) {
+      event.currentTarget.focus();
+    }
+
+    if (onClick) {
+      onClick(event);
+    }
+  };
+
+  return (
+    <EnhancedDateRangePickerDayContainer>
+      <EnhancedDateRangePickerDayRoot
+        ref={handleRef}
+        centerRipple
+        data-testid="day"
+        disabled={disabled}
+        tabIndex={selected ? 0 : -1}
+        onKeyDown={(event) => onKeyDown(event, day)}
+        onFocus={(event) => onFocus(event, day)}
+        onBlur={(event) => onBlur(event, day)}
+        onMouseEnter={(event) => onMouseEnter(event, day)}
+        onClick={handleClick}
+        onMouseDown={handleMouseDown}
+        draggable={draggable}
+        {...other}
+        ownerState={ownerState}
+        className={clsx(classes.root, className)}
+      >
+        {!children ? utils.format(day, 'dayOfMonth') : children}
```

Comment:
Swap the ternary
Avoid `!a ? b : c`

Here you could even use:

```suggestion
        {children ?? utils.format(day, 'dayOfMonth')}
```

And if you want an explicit `children={null}` to work, you do need the ternay (be careful when using `!`):

```suggestion
        {children === undefined ? utils.format(day, 'dayOfMonth') : children}
```

Reply:
Thanks 👌 Changed it in `PickersDay.tsx` too